### PR TITLE
check(envoy): min prerequisites for existence of an envoy sidecar

### PR DESCRIPTION
**Pull request by SanyaKochhar**
* Renames envoy sidecar check to be more explicit and extracts getConfigurator into kubernetesHelper
* Adds checks for min num containers existing in a pod
* Adds check for a pod having a valid proxy UUID label

### CODE REVIEW

**Comment by SanyaKochhar** [comment]
Ref: https://github.com/openservicemesh/osm-health/pull/3#discussion_r674383426

> shashankram 9 days ago
> hasEnvoySidecar() needs to include a few checks, all of which are not going to ensure 100% correctness, but should suffice to start with:
> 1. verify the pod is meshed. This can be identified by inspecting the pod's label, and ensuring it has at least 3 containers (app, init, sidecar)
> 2. Verify that the non-app containers correspond to 1. osm-init, 2. envoy

Currently, following format of other checks, this PR has 3 checks:
1. HasExpectedEnvoyImage: envoy sidecar image exists
2. HasMinExpectedContainers: at least 3 containers exist
3. HasProxyUUIDLabel: envoy uuid in container i.e., there has at some point been sidecar injection

I'm thinking this could this be collapsed into one EnvoyContainerMinRequisitesCheck? Is this informative enough to a user as a single check?

File: pkg/connectivity/pod-to-pod.go [hunk 1]
```diff
@@ -25,12 +25,16 @@ func PodToPod(fromPod *v1.Pod, toPod *v1.Pod) common.Result {
...
-		pod.HasEnvoySidecar(fromPod),
+		pod.HasMinExpectedContainers(fromPod, 3),
+		pod.HasExpectedEnvoyImage(fromPod),
+		pod.HasProxyUUIDLabel(fromPod),
 
 		// Check destination Pod's namespace
 		namespace.IsInjectEnabled(client, toPod.Namespace),
 		namespace.IsMonitoredBy(client, toPod.Namespace, meshName),
-		pod.HasEnvoySidecar(toPod),
+		pod.HasMinExpectedContainers(toPod, 3),
```
Comment (draychev) [inline pkg/connectivity/pod-to-pod.go:35]: It seems to me that this should be 2 containers (Envoy + Payload).  And then maybe we can have `pod.HasInitContainer`

(And then on top of that we need to check if the init container sets up IPTables etc.)

**Review by draychev — COMMENTED** [review]
This is directionally correct! 
Thank you @SanyaKochhar

File: pkg/connectivity/pod-to-pod.go (hunk 1 above)
Comment (SanyaKochhar) [reply pkg/connectivity/pod-to-pod.go:35]: Going to do this in a follow up PR! Having some trouble with the HasInitContainer check failing

**Review by draychev — APPROVED** [review]

**Review by johnsonshi — APPROVED** [review]

**Pull request merged**